subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -235,18 +239,52 @@ public void setForkOptions(ForkOptions forkOptions) {
 
     /**
      * Returns the bootstrap classpath to be used for the compiler process. Defaults to {@code null}.
+     *
+     * @deprecated Use {@link #getBootstrapClasspath()} instead.
      */
-    @Input
-    @Optional
+    @Deprecated
+    @Internal
     public String getBootClasspath() {
+        if (bootClasspath == null && bootstrapClasspath != null) {
+            return bootstrapClasspath.getAsPath();
+        }
         return bootClasspath;
     }
 
     /**
      * Sets the bootstrap classpath to be used for the compiler process. Defaults to {@code null}.
+     *
+     * @deprecated Use {@link #setBootstrapClasspath(FileCollection)} instead.
      */
+    @Deprecated
     public void setBootClasspath(String bootClasspath) {
         this.bootClasspath = bootClasspath;
+        this.bootstrapClasspath = null;
+    }
+
+    /**
+     * Returns the bootstrap classpath to be used for the compiler process. Defaults to {@code null}.
+     *
+     * @since 4.3
+     */
+    // TODO This could be a @CompileClasspath, but that would require a lot of processing
```

Comment:
Did you finally try out what the overhead of using `@CompileClasspath` is? If the overhead is too big, I would suggest removing the `TODO`.

Reply:
Done, updated the comments here, too.